oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profiling/trace/ProfileThreadSnapshotRecord.java

```diff
@@ -69,6 +70,42 @@ public class ProfileThreadSnapshotRecord extends Record {
     private int sequence;
     @Column(name = STACK_BINARY)
     private byte[] stackBinary;
+    @ElasticSearch.EnableDocValues
+    @Column(name = LANGUAGE_TYPE)
+    @BanyanDB.NoIndexing
+    private int languageType; // store as 0/1 for storage compatibility
+
+    public enum Language {
```

Comment:
You should not use a public non-static enum. Please move this to a separate class.